tensorflow/compiler/jit/xla_compilation_cache.h

```diff
@@ -189,6 +214,30 @@ class XlaCompilationCache : public ResourceBase {
   absl::flat_hash_map<string, ClusterCompileStats> cluster_compile_stats_
       TF_GUARDED_BY(cluster_compile_stats_mu_);
 
+  struct AsyncCompilation {
+    mutex async_compilation_mu_;
+
+    // Number of threads for asynchronous compilations.
+    static constexpr int64 kNumCompilerThreads = 10;
+
+    // Maximum number of ongoing compilations.
+    static constexpr int64 kMaxNumOngoingCompilations = kNumCompilerThreads;
+
+    // Number of ongoing compilations.
+    int64 num_ongoing_compilations GUARDED_BY(async_compilation_mu_) = 0;
+
+    // Pool of threads for asynchronous compilations.
+    std::unique_ptr<thread::ThreadPool> compiler_threads;
+
+    AsyncCompilation()
+    {
+       compiler_threads = absl::make_unique<tensorflow::thread::ThreadPool>(
+         tensorflow::Env::Default(), "aync_compiler_threads",
+         kNumCompilerThreads);
+    }
+
+  } async_compilation_;
```

Comment:
Same here, `async_compilation_state_` looks a bit better to me.

Reply:
done